### ISSUE TITLE
real double buffering

### ISSUE THREAD

**Issue by stapelberg** [i3/i3#661]
At the moment, we set pixmaps as background for the different windows. There still is some flickering when switching cons in a stack. Maybe this is gone when we use real double-buffering (render to a second pixmap, switch pixmaps)? Also see if the X double buffer extension can help us, though probably not (not available on all systems I guess).

**Comment by lindhe** [comment]
I experience extensive tearing, particularly with youtube videos. Not everytime, but it is a sporadic and frequent problem for me. I've tried installing compton and running `compton --backend glx --vsync opengl` but there is no difference. Is there anything else I should do to get rid of the tearing?

**Comment by Airblader** [comment]
Did you make sure you're running the latest driver for your graphics card?

**Comment by lindhe** [comment]
Yes. But, I just narrowed down the problem, and it might not be i3's fault after all. The problem was twofold: i3 without compton causes tearing, but so does driving [two monitors with different orientation with xrandr](https://wiki.archlinux.org/index.php/NVIDIA#Screen_Tearing_with_Multiple_Monitor_Orientations), at least with the nvidia driver.

When using the same orientation and running compton the tearing problem was fixed.

I guess this is an xrandr bug. Or nvidia, hard to say. Does anyone know? According to the Arch wiki there does not seem to be another fix than using the same orientation.

So it does not seem to be because of i3, but since i3 depends so heavily on xrandr it might be worth looking to solve it. Any ideas?

**Comment by Airblader** [comment]
My guess would be a driver issue, which you could more or less easily verify by using nouveau to see if it has the same problem. i3 not being at fault here was pretty clear since we don't render clients. ;)

**Comment by lindhe** [comment]
Trying different drivers will be a mission for a day that is not the day before a math exam. I guess you are right, but I'll get back as soon as I can comfirm this.

**Comment by andrei-korshikov** [comment]
`TearFree` option of `modesetting` driver works fine, so I don't use picom/compton at all. More info:
* https://www.phoronix.com/news/xf86-video-modesetting-TearFree
* https://wiki.archlinux.org/title/Intel_graphics#With_the_modesetting_driver_2
